simd/src/Kokkos_SIMD.hpp

```diff
@@ -99,83 +110,143 @@ struct ForSpace;
 #ifdef KOKKOS_ENABLE_SERIAL
 template <>
 struct ForSpace<Kokkos::Serial> {
-  using type = host_native;
+  using type = host_fixed_native;
+
+  template <int N>
+  using simd_abi = host_native_abi<N>;
 };
 #endif
 
 #ifdef KOKKOS_ENABLE_CUDA
 template <>
 struct ForSpace<Kokkos::Cuda> {
   using type = scalar;
+
+  template <int N>
+  using simd_abi = scalar;
 };
 #endif
 
 #ifdef KOKKOS_ENABLE_THREADS
 template <>
 struct ForSpace<Kokkos::Threads> {
-  using type = host_native;
+  using type = host_fixed_native;
+
+  template <int N>
+  using simd_abi = host_native_abi<N>;
 };
 #endif
 
 #ifdef KOKKOS_ENABLE_HPX
 template <>
 struct ForSpace<Kokkos::Experimental::HPX> {
   using type = scalar;
+
+  template <int N>
+  using simd_abi = scalar;
 };
 #endif
 
 #ifdef KOKKOS_ENABLE_OPENMP
 template <>
 struct ForSpace<Kokkos::OpenMP> {
-  using type = host_native;
+  using type = host_fixed_native;
+
+  template <int N>
+  using simd_abi = host_native_abi<N>;
 };
 #endif
 
 #ifdef KOKKOS_ENABLE_OPENMPTARGET
 template <>
 struct ForSpace<Kokkos::Experimental::OpenMPTarget> {
   using type = scalar;
+
+  template <int N>
+  using simd_abi = scalar;
 };
 #endif
 
 #ifdef KOKKOS_ENABLE_OPENACC
 template <>
 struct ForSpace<Kokkos::Experimental::OpenACC> {
   using type = scalar;
+
+  template <int N>
+  using simd_abi = scalar;
 };
 #endif
 
 #ifdef KOKKOS_ENABLE_HIP
 template <>
 struct ForSpace<Kokkos::HIP> {
   using type = scalar;
+
+  template <int N>
+  using simd_abi = scalar;
 };
 #endif
 
 #ifdef KOKKOS_ENABLE_SYCL
 template <>
 struct ForSpace<Kokkos::SYCL> {
   using type = scalar;
+
+  template <int N>
+  using simd_abi = scalar;
 };
 #endif
 
+template <class Space = Kokkos::DefaultExecutionSpace>
+using native_fixed_abi = typename ForSpace<Space>::type;
+
+template <int N, class Space = Kokkos::DefaultExecutionSpace>
+using native_abi = typename ForSpace<Space>::template simd_abi<N>;
+
 }  // namespace Impl
 
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
```

Comment:
Instead of this guard, could we just add `KOKKOS_DEPRRECATED` for the `using` statements like you do below?

Reply:
That was my initial approach, but placing the `deprecated` attributes on these aliases would cause every simd compilation to output deprecation errors, because both `simd_abi::ForSpace` and `simd_abi::native` are used for `native_simd` and `native_simd_mask`, which are now also deprecated.